Add automation for word docs, analgous to notebooks

hamelsmu opened the pull request:
I promised that I would also do a Word converter in our email discussion.  So, I'm opening this PR to fulfill my promises.  

## Summary of changes made:

- Rearranged `_action_files` such that word and notebooks both share the same naming function, and moved shared code into `_action_files/fast_template.py`
- Added Action Workflow that converts word to md
- Added scaffolding to `_config.yml` to guide people to specify their `url`, which is useful when someone hosts their blog as not root (for example a project site, or custom domain).  This value being null does not break anything.
- Added a `_notebooks/README.md` file, in a similar spirit that was done for notebooks. 

## Key Differences from notebook converter

- I do NOT have the requirement of having the magic keyword `/sync` in commit messages for word.  I thought that is too technical and wanted to keep it simple, assuming many people would just use the sync feature of GitHub Desktop.

- I'm injecting `{{site.url}}` into image links, and saving media under `/assets/img`.  I did this so people do not have to fiddle with moving images around or renaming links.

Please let me know if this interface is suitable for users, happy to iterate it on it as you suggest 🙇

hamelsmu committed:
refactor action files

hamelsmu committed:
draft word2post workflow

hamelsmu committed:
temporarily add test notebooks

hamelsmu committed:
temporarily add word docs for testing

hamelsmu committed:
troubleshoot workflow

hamelsmu committed:
troubleshoot workflow

hamelsmu committed:
corrected path names

hamelsmu committed:
see pandoc version

hamelsmu committed:
see pandoc version

hamelsmu committed:
see pandoc version

hamelsmu committed:
test jupyter converter

hamelsmu committed:
/sync

hamelsmu committed:
/sync stuff

hamelsmu committed:
correct workflow syntax

hamelsmu committed:
Merge pull request #3 from hamelsmu/word-processor

Word processor

Commit:
Update MS Word Blog Posts

Commit:
Update Jupyter Blog Posts

hamelsmu committed:
Delete test-notebook.ipynb

hamelsmu committed:
Delete test-notebook-2.ipynb

hamelsmu committed:
Delete Test Word Post-Two.docx

hamelsmu committed:
Delete Test Word Post.docx

hamelsmu committed:
correct workflow syntax

hamelsmu committed:
Merge pull request #4 from hamelsmu/word-processor

correct workflow syntax

hamelsmu committed:
correct workflow syntax /sync

hamelsmu committed:
Merge pull request #5 from hamelsmu/word-processor

correct workflow syntax /sync

hamelsmu committed:
add files

hamelsmu committed:
added folder

hamelsmu committed:
Merge branch 'master' of https://github.com/hamelsmu/fast_template

hamelsmu committed:
add docs

hamelsmu committed:
update nb

hamelsmu committed:
update docs

hamelsmu committed:
removed posts

hamelsmu committed:
/sync stuff

hamelsmu committed:
Merge pull request #6 from hamelsmu/word-processor

Word processor

Commit:
Update MS Word Blog Posts

hamelsmu committed:
remove posts

hamelsmu committed:
see if relative directory works

hamelsmu committed:
Merge branch 'word-processor'

hamelsmu committed:
remove fies

hamelsmu committed:
/sync

hamelsmu committed:
/sync notebooks

hamelsmu committed:
Merge pull request #8 from hamelsmu/word-processor

/sync

Commit:
Update MS Word Blog Posts

Commit:
Update Jupyter Blog Posts

hamelsmu committed:
Update 2020-01-27-Test-Word-Post-Two.md

hamelsmu committed:
Update 2020-01-27-Test-Word-Post.md

hamelsmu committed:
Update 2020-01-27-Test-Word-Post.md

hamelsmu committed:
Update 2020-01-27-Test-Word-Post.md

hamelsmu committed:
Update 2020-01-27-Test-Word-Post.md

hamelsmu committed:
Update _config.yml

hamelsmu committed:
Update 2020-01-27-Test-Word-Post.md

hamelsmu committed:
Update 2020-01-27-Test-Word-Post.md

hamelsmu committed:
Update _config.yml

hamelsmu committed:
Update _config.yml

hamelsmu committed:
/sync word files

hamelsmu committed:
Merge pull request #9 from hamelsmu/word-processor

/sync word files

hamelsmu committed:
correct syntax

hamelsmu committed:
/sync!

hamelsmu committed:
Merge pull request #10 from hamelsmu/word-processor

Word processor

Commit:
Update Jupyter Blog Posts

hamelsmu committed:
/sync test

hamelsmu committed:
removed extreanous files

hamelsmu committed:
Merge pull request #11 from hamelsmu/word-processor

Word processor

Commit:
Update MS Word Blog Posts

hamelsmu committed:
Update _config.yml

hamelsmu committed:
Update _config.yml

/sync

Commit:
Update MS Word Blog Posts

Commit:
Update Jupyter Blog Posts

hamelsmu committed:
update word2post

hamelsmu committed:
Merge branch 'master' of https://github.com/hamelsmu/fast_template

hamelsmu committed:
remove /sync requirement

Commit:
Update Jupyter Blog Posts

hamelsmu committed:
remove demo files

hamelsmu committed:
Merge branch 'master' of https://github.com/hamelsmu/fast_template

hamelsmu committed:
cleanup branch

hamelsmu committed:
more cleanup

hamelsmu committed:
nit end of file

hamelsmu committed:
nit end of file

jph00 commented on _config.yml:
```diff
@@ -9,6 +9,7 @@
 title: Edit _config.yml to set your title!
 description: This is where the description of your site will go. You should change it by editing the _config.yml file. It can be as long as you like! Happy blogging... ❤
 github_username: jph00
+url: # the full url for your site for your site, e.g. http://username.github.io/blog
```
This is optional, at least in some cases, right? If so, move it after email and twitter, and add a comment above saying it's optional, and explain when it's needed.

Might it be possible to use the metadata from this plugin instead?: https://github.com/jekyll/github-metadata . I've used it for the social icons already.

jph00 reviewed (changes requested):
This is looking great!

hamelsmu committed:
test plugin /sync

hamelsmu committed:
added word doc

Commit:
Update MS Word Blog Posts

hamelsmu committed:
update link

hamelsmu committed:
Merge branch 'master' of https://github.com/hamelsmu/fast_template

Commit:
Update MS Word Blog Posts

hamelsmu committed:
troubleshoot

Commit:
Update MS Word Blog Posts

hamelsmu committed:
test _config

hamelsmu committed:
Merge branch 'master' of https://github.com/hamelsmu/fast_template

hamelsmu committed:
remove files for testing

hamelsmu committed:
remove generated test

hamelsmu committed:
nit - clean up comments

hamelsmu committed:
remove testing media artifacts

hamelsmu committed:
spell / grammar check on README

hamelsmu commented:
@jph00 Great idea.  I made the following changes:

- Moved the scaffolding below email / twitter ( it is completely commented out, doesn't work well when it is null ) 
- Added the plugin to the `Gemfile` & `_config.yaml`
- Updated scripts to use `{{ site.url }}{{ site.baseurl }}` for image links, instead of just `{{ site.url }}`to reflect the new plugin 🎉 
- Updated the README to mention setting the parameters `url` and `baseurl` ONLY in the case of a custom domain
- Now users don't have to set anything in `_config.yaml` when using a project (non-root) pages, b/c of the cool plugin 🚀 

I tested it to make sure it works, works great!

jph00 commented:
Great!
